payments-core/src/main/java/com/stripe/android/StripePaymentController.kt

```diff
@@ -586,11 +597,44 @@ internal class StripePaymentController internal constructor(
         }
     }
 
+    /**
+     * Decide whether a [StripeIntent] confirmation request's returnUrl should be overridden
+     * to [DefaultReturnUrl] if client doesn't set any value.
+     */
+    private suspend fun shouldOverrideReturnUrl(
+        confirmStripeIntentParams: ConfirmStripeIntentParams,
+        requestOptions: ApiRequest.Options
+    ): Boolean {
+        // Get payment method type if it's set in confirmStripeIntentParams,
+        // otherwise get the type by fetching the StripeIntent
+        val paymentMethodType =
+            when (confirmStripeIntentParams) {
+                is ConfirmPaymentIntentParams -> {
+                    confirmStripeIntentParams.paymentMethodCreateParams
+                }
+                is ConfirmSetupIntentParams -> {
+                    confirmStripeIntentParams.paymentMethodCreateParams
+                }
+            }?.type?.paymentMethodType ?: run {
+                stripeRepository.retrieveStripeIntent(
+                    confirmStripeIntentParams.clientSecret,
+                    requestOptions,
+                    EXPAND_PAYMENT_METHOD
+                ).paymentMethod?.type
+            }
```

Comment:
What if the user is confirming with an existing PaymentMethod (i.e. using `paymentMethodId`) that would normally be in the blocklist? This logic wouldn't cover that case.

Reply:
it seems this is a blocker we just can't do from client side - I tried to [update the PI](https://stripe.com/docs/api/payment_intents/update) with the pm or [retrieve the PM](https://stripe.com/docs/api/payment_methods/retrieve), both would require secret key